modules/zio-config/src/main/scala/pureconfig/module/zioconfig/package.scala

```diff
@@ -0,0 +1,22 @@
+package pureconfig.module
+
+import com.typesafe.config.{Config, ConfigFactory}
+import zio.config._
+import zio.config.typesafe._
+
+import pureconfig.ConfigConvert
+
+package object zioconfig {
+  implicit def zioConfigConvert[A](implicit cd: ConfigDescriptor[A]): ConfigConvert[A] =
+    ConfigConvert[Config].xemap(
+      c =>
+        TypesafeConfigSource
+          .fromConfig(c)
+          .flatMap(cs => read(cd.from(cs)))
+          .left
+          .map(ZioConfigReadError.apply),
+      //`zio-config` write should not fail but is allowed to fail
+      //defaulting write failures to empty `Config` so it fits `pureconfig` writer
+      _.toHocon(cd).map(_.toConfig).getOrElse(ConfigFactory.empty)
```

Comment:
One thought: should we throw an exception in case of an error instead of returning an empty object? Write failures may be unexpected and even technically impossible, but if some error ends up occurring I'd rather fail fast and loud rather than using a default that may make things very confusing (it may even be a valid success result in some cases). What are your thoughts?

Reply:
@ruippeixotog I understand the issue of silent failures but on the other hand users might be depending on `pureconfig` writes to be safe and pure. I will switch to throwing to conform to what has been done in other modules for write errors, e.g. the `shapeless` writers in `pureconfig-generic`.

`zio-config` writes can still be made to fail for arbitrary reasons, e.g:
```scala
val confDescChild: ConfigDescriptor[Person] =
(string("name") |@| int("age") |@| list("children")(confDescChild))(Person.apply, Person.unapply)
  .transformOrFailRight(identity, {
    case Person(_, age, _) if age < 18 => Left("Too old to be a child.")
    case Person(_, _, children) if children.nonEmpty => Left("Child cannot have children.")
    case p => Right(p)
  })
}
```
Although personally, in cases like that, I would prefer refining the types. Maybe I should also add that as another example for write failures.